tiatoolbox/cli.py

```diff
@@ -434,5 +439,94 @@ def tissue_mask(
             )
 
 
+@main.command()
+@click.option(
+    "--predefined_model",
+    help="predefined model used to process the data. the format is "
+    "<model_name>_<dataset_trained_on>. For example, `resnet18-kather100K` "
+    "is a resnet18 model trained on the kather dataset.",
+    default="resnet18-kather100K",
+)
+@click.option(
+    "--pretrained_weight",
+    help="path to the model weight file. If not supplied, the default "
+    "pretrained weight will be used.",
+    default=None,
+)
+@click.option(
+    "--img_input",
+    help="path to the input directory containing images to process or an "
+    "individual file.",
+)
+@click.option(
+    "--output_path",
+    help="output directory where model predictions will be saved.",
+    default="patch_prediction",
+)
+@click.option(
+    "--batch_size",
+    help="number of images to feed into the model each time.",
+    default=16,
+)
+@click.option(
+    "--return_probs",
+    help="whether to return raw model probabilities.",
+    default=False,
+)
+@click.option(
+    "--file_types",
+    help="file types to capture from directory. "
+    "default='*.png', '*.jpg', '*.jpeg', '*.tif', '*.tiff'",
+    default="*.png, *.jpg, *.jpeg, *.tif, *.tiff",
+)
+def patch_predictor(
+    predefined_model,
+    pretrained_weight,
+    img_input,
+    output_path,
+    batch_size,
+    return_probs,
+    file_types,
+):
+    """Process an image/directory of input images with a patch classification CNN."""
+    file_types = tuple(file_types.split(", "))
+    output_path = pathlib.Path(output_path)
+
+    if os.path.isdir(img_input):
+        img_files = utils.misc.grab_files_from_dir(
+            input_path=img_input, file_types=file_types
+        )
+    elif os.path.isfile(img_input):
+        img_files = [
+            img_input,
+        ]
+    else:
+        raise FileNotFoundError
+
+    if predefined_model.lower() not in __pretrained_model:
+        raise ValueError("Predefined model `%s` does not exist." % predefined_model)
+
+    if len(img_files) < batch_size:
+        batch_size = len(img_files)
+
+    dataset = Patch_Dataset(img_files)
+
+    predictor = CNN_Patch_Predictor(
+        predefined_model=predefined_model,
+        pretrained_weight=pretrained_weight,
+        batch_size=batch_size,
+    )
+
+    output = predictor.predict(dataset, return_probs=return_probs, on_gpu=False)
+
+    output_file_path = os.path.join(output_path, "results.json")
+    if not output_path.is_dir():
+        os.makedirs(output_path)
+    print(output)
+    print(output_file_path)
+    # with open(output_file_path, "w") as handle:
+    #     json.dump(output, handle)
```

Comment:
Can this comment be removed?

Reply:
Oop, this should be added back rather than commented out.